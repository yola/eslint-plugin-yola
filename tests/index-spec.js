const plugin = require('../index');

describe('eslint-plugin-yola', () => {
  it('must contain configs and rules properties', () => {
    expect(plugin).toHaveProperty('configs.base.rules');
    expect(plugin).toHaveProperty('configs.react.rules');
    expect(plugin).toHaveProperty('rules');
  });
});
