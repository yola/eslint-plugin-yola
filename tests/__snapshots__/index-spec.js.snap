// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`eslint-plugin-yola must contain configs and rules properties 1`] = `
Object {
  "base": Object {
    "env": Object {
      "browser": true,
      "es6": true,
      "node": true,
    },
    "overrides": Array [
      Object {
        "env": Object {
          "mocha": true,
        },
        "files": Array [
          "*-spec.js",
          "*-spec.jsx",
          "*.spec.js",
          "*.spec.jsx",
        ],
        "rules": Object {
          "import/no-extraneous-dependencies": Array [
            "error",
            Object {
              "devDependencies": true,
            },
          ],
        },
      },
    ],
    "parser": true,
    "parserOptions": Object {
      "ecmaFeatures": Object {
        "experimentalObjectRestSpread": true,
        "generators": false,
        "objectLiteralDuplicateProperties": false,
      },
      "ecmaVersion": 6,
      "sourceType": "module",
    },
    "plugins": Array [
      "yola",
    ],
    "rules": Object {
      "accessor-pairs": true,
      "array-bracket-newline": true,
      "array-bracket-spacing": true,
      "array-callback-return": true,
      "array-element-newline": true,
      "arrow-body-style": true,
      "arrow-parens": true,
      "arrow-spacing": true,
      "block-scoped-var": true,
      "block-spacing": true,
      "brace-style": true,
      "callback-return": true,
      "camelcase": true,
      "capitalized-comments": true,
      "class-methods-use-this": true,
      "comma-dangle": true,
      "comma-spacing": true,
      "comma-style": true,
      "complexity": true,
      "computed-property-spacing": true,
      "consistent-return": true,
      "consistent-this": true,
      "constructor-super": true,
      "curly": true,
      "default-case": true,
      "dot-location": true,
      "dot-notation": true,
      "eol-last": true,
      "eqeqeq": true,
      "for-direction": true,
      "func-call-spacing": true,
      "func-name-matching": true,
      "func-names": true,
      "func-style": true,
      "function-paren-newline": true,
      "generator-star-spacing": true,
      "getter-return": true,
      "global-require": true,
      "guard-for-in": true,
      "handle-callback-err": true,
      "id-blacklist": true,
      "id-length": true,
      "id-match": true,
      "indent": true,
      "init-declarations": true,
      "jsx-quotes": true,
      "key-spacing": true,
      "keyword-spacing": true,
      "line-comment-position": true,
      "linebreak-style": true,
      "lines-around-comment": true,
      "lines-around-directive": true,
      "lines-between-class-members": true,
      "max-depth": true,
      "max-len": true,
      "max-lines": true,
      "max-nested-callbacks": true,
      "max-params": true,
      "max-statements": true,
      "max-statements-per-line": true,
      "multiline-comment-style": true,
      "multiline-ternary": true,
      "new-cap": true,
      "new-parens": true,
      "newline-after-var": true,
      "newline-before-return": true,
      "newline-per-chained-call": true,
      "no-alert": true,
      "no-array-constructor": true,
      "no-await-in-loop": true,
      "no-bitwise": true,
      "no-buffer-constructor": true,
      "no-caller": true,
      "no-case-declarations": true,
      "no-catch-shadow": true,
      "no-class-assign": true,
      "no-compare-neg-zero": true,
      "no-cond-assign": true,
      "no-confusing-arrow": true,
      "no-console": true,
      "no-const-assign": true,
      "no-constant-condition": true,
      "no-continue": true,
      "no-control-regex": true,
      "no-debugger": true,
      "no-delete-var": true,
      "no-div-regex": true,
      "no-dupe-args": true,
      "no-dupe-class-members": true,
      "no-dupe-keys": true,
      "no-duplicate-case": true,
      "no-duplicate-imports": true,
      "no-else-return": true,
      "no-empty": true,
      "no-empty-character-class": true,
      "no-empty-function": true,
      "no-empty-pattern": true,
      "no-eq-null": true,
      "no-eval": true,
      "no-ex-assign": true,
      "no-extend-native": true,
      "no-extra-bind": true,
      "no-extra-boolean-cast": true,
      "no-extra-label": true,
      "no-extra-parens": true,
      "no-extra-semi": true,
      "no-fallthrough": true,
      "no-floating-decimal": true,
      "no-func-assign": true,
      "no-global-assign": true,
      "no-implicit-coercion": true,
      "no-implicit-globals": true,
      "no-implied-eval": true,
      "no-inline-comments": true,
      "no-inner-declarations": true,
      "no-invalid-regexp": true,
      "no-invalid-this": true,
      "no-irregular-whitespace": true,
      "no-iterator": true,
      "no-label-var": true,
      "no-labels": true,
      "no-lone-blocks": true,
      "no-lonely-if": true,
      "no-loop-func": true,
      "no-magic-numbers": true,
      "no-mixed-operators": true,
      "no-mixed-requires": true,
      "no-mixed-spaces-and-tabs": true,
      "no-multi-assign": true,
      "no-multi-spaces": true,
      "no-multi-str": true,
      "no-multiple-empty-lines": true,
      "no-native-reassign": true,
      "no-negated-condition": true,
      "no-negated-in-lhs": true,
      "no-nested-ternary": true,
      "no-new": true,
      "no-new-func": true,
      "no-new-object": true,
      "no-new-require": true,
      "no-new-symbol": true,
      "no-new-wrappers": true,
      "no-obj-calls": true,
      "no-octal": true,
      "no-octal-escape": true,
      "no-param-reassign": true,
      "no-path-concat": true,
      "no-plusplus": true,
      "no-process-env": true,
      "no-process-exit": true,
      "no-proto": true,
      "no-prototype-builtins": true,
      "no-redeclare": true,
      "no-regex-spaces": true,
      "no-restricted-globals": true,
      "no-restricted-imports": true,
      "no-restricted-modules": true,
      "no-restricted-properties": true,
      "no-restricted-syntax": true,
      "no-return-assign": true,
      "no-return-await": true,
      "no-script-url": true,
      "no-self-assign": true,
      "no-self-compare": true,
      "no-sequences": true,
      "no-shadow": true,
      "no-shadow-restricted-names": true,
      "no-spaced-func": true,
      "no-sparse-arrays": true,
      "no-sync": true,
      "no-tabs": true,
      "no-template-curly-in-string": true,
      "no-ternary": true,
      "no-this-before-super": true,
      "no-throw-literal": true,
      "no-trailing-spaces": true,
      "no-undef": true,
      "no-undef-init": true,
      "no-undefined": true,
      "no-underscore-dangle": true,
      "no-unexpected-multiline": true,
      "no-unmodified-loop-condition": true,
      "no-unneeded-ternary": true,
      "no-unreachable": true,
      "no-unsafe-finally": true,
      "no-unsafe-negation": true,
      "no-unused-expressions": true,
      "no-unused-labels": true,
      "no-unused-vars": true,
      "no-use-before-define": true,
      "no-useless-call": true,
      "no-useless-computed-key": true,
      "no-useless-concat": true,
      "no-useless-constructor": true,
      "no-useless-escape": true,
      "no-useless-rename": true,
      "no-useless-return": true,
      "no-var": true,
      "no-void": true,
      "no-warning-comments": true,
      "no-whitespace-before-property": true,
      "no-with": true,
      "nonblock-statement-body-position": true,
      "object-curly-newline": true,
      "object-curly-spacing": true,
      "object-property-newline": true,
      "object-shorthand": true,
      "one-var": true,
      "one-var-declaration-per-line": true,
      "operator-assignment": true,
      "operator-linebreak": true,
      "padded-blocks": true,
      "padding-line-between-statements": true,
      "prefer-arrow-callback": true,
      "prefer-const": true,
      "prefer-destructuring": true,
      "prefer-numeric-literals": true,
      "prefer-promise-reject-errors": true,
      "prefer-reflect": true,
      "prefer-rest-params": true,
      "prefer-spread": true,
      "prefer-template": true,
      "quote-props": true,
      "quotes": true,
      "radix": true,
      "require-await": true,
      "require-jsdoc": true,
      "require-yield": true,
      "rest-spread-spacing": true,
      "semi": true,
      "semi-spacing": true,
      "semi-style": true,
      "sort-imports": true,
      "sort-keys": true,
      "sort-vars": true,
      "space-before-blocks": true,
      "space-before-function-paren": true,
      "space-in-parens": true,
      "space-infix-ops": true,
      "space-unary-ops": true,
      "spaced-comment": true,
      "strict": true,
      "switch-colon-spacing": true,
      "symbol-description": true,
      "template-curly-spacing": true,
      "template-tag-spacing": true,
      "unicode-bom": true,
      "use-isnan": true,
      "valid-jsdoc": true,
      "valid-typeof": true,
      "vars-on-top": true,
      "wrap-iife": true,
      "wrap-regex": true,
      "yield-star-spacing": true,
      "yoda": true,
      "yola/import/default": true,
      "yola/import/export": true,
      "yola/import/extensions": true,
      "yola/import/first": true,
      "yola/import/imports-first": true,
      "yola/import/max-dependencies": true,
      "yola/import/named": true,
      "yola/import/namespace": true,
      "yola/import/newline-after-import": true,
      "yola/import/no-absolute-path": true,
      "yola/import/no-amd": true,
      "yola/import/no-anonymous-default-export": true,
      "yola/import/no-commonjs": true,
      "yola/import/no-deprecated": true,
      "yola/import/no-duplicates": true,
      "yola/import/no-dynamic-require": true,
      "yola/import/no-extraneous-dependencies": true,
      "yola/import/no-internal-modules": true,
      "yola/import/no-mutable-exports": true,
      "yola/import/no-named-as-default": true,
      "yola/import/no-named-as-default-member": true,
      "yola/import/no-named-default": true,
      "yola/import/no-namespace": true,
      "yola/import/no-nodejs-modules": true,
      "yola/import/no-restricted-paths": true,
      "yola/import/no-unassigned-import": true,
      "yola/import/no-unresolved": true,
      "yola/import/no-webpack-loader-syntax": true,
      "yola/import/order": true,
      "yola/import/prefer-default-export": true,
      "yola/import/unambiguous": true,
    },
    "settings": Object {
      "import/core-modules": Array [],
      "import/extensions": Array [
        ".js",
        ".jsx",
      ],
      "import/ignore": Array [
        "node_modules",
        "\\\\.(coffee|scss|css|less|hbs|svg|json)$",
      ],
      "import/resolver": Object {
        "node": Object {
          "extensions": Array [
            ".js",
            ".json",
          ],
        },
      },
    },
  },
  "react": Object {
    "env": Object {
      "browser": true,
    },
    "overrides": Array [
      Object {
        "env": Object {
          "mocha": true,
        },
        "files": Array [
          "*-spec.js",
          "*-spec.jsx",
          "*.spec.js",
          "*.spec.jsx",
        ],
        "rules": Object {
          "import/no-extraneous-dependencies": Array [
            "error",
            Object {
              "devDependencies": true,
            },
          ],
        },
      },
    ],
    "parser": true,
    "parserOptions": Object {
      "ecmaFeatures": Object {
        "experimentalObjectRestSpread": true,
        "jsx": true,
      },
      "ecmaVersion": 2017,
      "sourceType": "module",
    },
    "plugins": Array [
      "yola",
    ],
    "rules": Object {
      "class-methods-use-this": true,
      "jsx-quotes": true,
      "strict": true,
      "yola/jsx-a11y/accessible-emoji": true,
      "yola/jsx-a11y/alt-text": true,
      "yola/jsx-a11y/anchor-has-content": true,
      "yola/jsx-a11y/anchor-is-valid": true,
      "yola/jsx-a11y/aria-activedescendant-has-tabindex": true,
      "yola/jsx-a11y/aria-props": true,
      "yola/jsx-a11y/aria-proptypes": true,
      "yola/jsx-a11y/aria-role": true,
      "yola/jsx-a11y/aria-unsupported-elements": true,
      "yola/jsx-a11y/click-events-have-key-events": true,
      "yola/jsx-a11y/heading-has-content": true,
      "yola/jsx-a11y/html-has-lang": true,
      "yola/jsx-a11y/iframe-has-title": true,
      "yola/jsx-a11y/img-redundant-alt": true,
      "yola/jsx-a11y/interactive-supports-focus": true,
      "yola/jsx-a11y/label-has-for": true,
      "yola/jsx-a11y/lang": true,
      "yola/jsx-a11y/media-has-caption": true,
      "yola/jsx-a11y/mouse-events-have-key-events": true,
      "yola/jsx-a11y/no-access-key": true,
      "yola/jsx-a11y/no-autofocus": true,
      "yola/jsx-a11y/no-distracting-elements": true,
      "yola/jsx-a11y/no-interactive-element-to-noninteractive-role": true,
      "yola/jsx-a11y/no-noninteractive-element-interactions": true,
      "yola/jsx-a11y/no-noninteractive-element-to-interactive-role": true,
      "yola/jsx-a11y/no-noninteractive-tabindex": true,
      "yola/jsx-a11y/no-onchange": true,
      "yola/jsx-a11y/no-redundant-roles": true,
      "yola/jsx-a11y/no-static-element-interactions": true,
      "yola/jsx-a11y/role-has-required-aria-props": true,
      "yola/jsx-a11y/role-supports-aria-props": true,
      "yola/jsx-a11y/scope": true,
      "yola/jsx-a11y/tabindex-no-positive": true,
      "yola/react/boolean-prop-naming": true,
      "yola/react/default-props-match-prop-types": true,
      "yola/react/display-name": true,
      "yola/react/forbid-component-props": true,
      "yola/react/forbid-elements": true,
      "yola/react/forbid-foreign-prop-types": true,
      "yola/react/forbid-prop-types": true,
      "yola/react/jsx-boolean-value": true,
      "yola/react/jsx-closing-bracket-location": true,
      "yola/react/jsx-closing-tag-location": true,
      "yola/react/jsx-curly-brace-presence": true,
      "yola/react/jsx-curly-spacing": true,
      "yola/react/jsx-equals-spacing": true,
      "yola/react/jsx-filename-extension": true,
      "yola/react/jsx-first-prop-new-line": true,
      "yola/react/jsx-handler-names": true,
      "yola/react/jsx-indent": true,
      "yola/react/jsx-indent-props": true,
      "yola/react/jsx-key": true,
      "yola/react/jsx-max-props-per-line": true,
      "yola/react/jsx-no-bind": true,
      "yola/react/jsx-no-comment-textnodes": true,
      "yola/react/jsx-no-duplicate-props": true,
      "yola/react/jsx-no-literals": true,
      "yola/react/jsx-no-target-blank": true,
      "yola/react/jsx-no-undef": true,
      "yola/react/jsx-pascal-case": true,
      "yola/react/jsx-sort-prop-types": true,
      "yola/react/jsx-sort-props": true,
      "yola/react/jsx-space-before-closing": true,
      "yola/react/jsx-tag-spacing": true,
      "yola/react/jsx-uses-react": true,
      "yola/react/jsx-uses-vars": true,
      "yola/react/jsx-wrap-multilines": true,
      "yola/react/no-array-index-key": true,
      "yola/react/no-children-prop": true,
      "yola/react/no-danger": true,
      "yola/react/no-danger-with-children": true,
      "yola/react/no-deprecated": true,
      "yola/react/no-did-mount-set-state": true,
      "yola/react/no-did-update-set-state": true,
      "yola/react/no-direct-mutation-state": true,
      "yola/react/no-find-dom-node": true,
      "yola/react/no-is-mounted": true,
      "yola/react/no-multi-comp": true,
      "yola/react/no-redundant-should-component-update": true,
      "yola/react/no-render-return-value": true,
      "yola/react/no-set-state": true,
      "yola/react/no-string-refs": true,
      "yola/react/no-typos": true,
      "yola/react/no-unescaped-entities": true,
      "yola/react/no-unknown-property": true,
      "yola/react/no-unused-prop-types": true,
      "yola/react/no-unused-state": true,
      "yola/react/no-will-update-set-state": true,
      "yola/react/prefer-es6-class": true,
      "yola/react/prefer-stateless-function": true,
      "yola/react/prop-types": true,
      "yola/react/react-in-jsx-scope": true,
      "yola/react/require-default-props": true,
      "yola/react/require-optimization": true,
      "yola/react/require-render-return": true,
      "yola/react/self-closing-comp": true,
      "yola/react/sort-comp": true,
      "yola/react/sort-prop-types": true,
      "yola/react/style-prop-object": true,
      "yola/react/void-dom-elements-no-children": true,
    },
    "settings": Object {
      "import/resolver": Object {
        "node": Object {
          "extensions": Array [
            ".js",
            ".jsx",
            ".json",
          ],
        },
      },
      "propWrapperFunctions": Array [
        "forbidExtraProps",
        "exact",
        "Object.freeze",
      ],
      "react": Object {
        "pragma": "React",
        "version": "15.0",
      },
    },
  },
}
`;

exports[`eslint-plugin-yola must contain configs and rules properties 2`] = `
Object {
  "import/default": true,
  "import/export": true,
  "import/exports-last": true,
  "import/extensions": true,
  "import/first": true,
  "import/imports-first": true,
  "import/max-dependencies": true,
  "import/named": true,
  "import/namespace": true,
  "import/newline-after-import": true,
  "import/no-absolute-path": true,
  "import/no-amd": true,
  "import/no-anonymous-default-export": true,
  "import/no-commonjs": true,
  "import/no-deprecated": true,
  "import/no-duplicates": true,
  "import/no-dynamic-require": true,
  "import/no-extraneous-dependencies": true,
  "import/no-internal-modules": true,
  "import/no-mutable-exports": true,
  "import/no-named-as-default": true,
  "import/no-named-as-default-member": true,
  "import/no-named-default": true,
  "import/no-namespace": true,
  "import/no-nodejs-modules": true,
  "import/no-restricted-paths": true,
  "import/no-unassigned-import": true,
  "import/no-unresolved": true,
  "import/no-webpack-loader-syntax": true,
  "import/order": true,
  "import/prefer-default-export": true,
  "import/unambiguous": true,
  "jsx-a11y/accessible-emoji": true,
  "jsx-a11y/alt-text": true,
  "jsx-a11y/anchor-has-content": true,
  "jsx-a11y/anchor-is-valid": true,
  "jsx-a11y/aria-activedescendant-has-tabindex": true,
  "jsx-a11y/aria-props": true,
  "jsx-a11y/aria-proptypes": true,
  "jsx-a11y/aria-role": true,
  "jsx-a11y/aria-unsupported-elements": true,
  "jsx-a11y/click-events-have-key-events": true,
  "jsx-a11y/heading-has-content": true,
  "jsx-a11y/html-has-lang": true,
  "jsx-a11y/iframe-has-title": true,
  "jsx-a11y/img-redundant-alt": true,
  "jsx-a11y/interactive-supports-focus": true,
  "jsx-a11y/label-has-for": true,
  "jsx-a11y/lang": true,
  "jsx-a11y/media-has-caption": true,
  "jsx-a11y/mouse-events-have-key-events": true,
  "jsx-a11y/no-access-key": true,
  "jsx-a11y/no-autofocus": true,
  "jsx-a11y/no-distracting-elements": true,
  "jsx-a11y/no-interactive-element-to-noninteractive-role": true,
  "jsx-a11y/no-noninteractive-element-interactions": true,
  "jsx-a11y/no-noninteractive-element-to-interactive-role": true,
  "jsx-a11y/no-noninteractive-tabindex": true,
  "jsx-a11y/no-onchange": true,
  "jsx-a11y/no-redundant-roles": true,
  "jsx-a11y/no-static-element-interactions": true,
  "jsx-a11y/role-has-required-aria-props": true,
  "jsx-a11y/role-supports-aria-props": true,
  "jsx-a11y/scope": true,
  "jsx-a11y/tabindex-no-positive": true,
  "react/boolean-prop-naming": true,
  "react/default-props-match-prop-types": true,
  "react/display-name": true,
  "react/forbid-component-props": true,
  "react/forbid-elements": true,
  "react/forbid-foreign-prop-types": true,
  "react/forbid-prop-types": true,
  "react/jsx-boolean-value": true,
  "react/jsx-closing-bracket-location": true,
  "react/jsx-closing-tag-location": true,
  "react/jsx-curly-brace-presence": true,
  "react/jsx-curly-spacing": true,
  "react/jsx-equals-spacing": true,
  "react/jsx-filename-extension": true,
  "react/jsx-first-prop-new-line": true,
  "react/jsx-handler-names": true,
  "react/jsx-indent": true,
  "react/jsx-indent-props": true,
  "react/jsx-key": true,
  "react/jsx-max-props-per-line": true,
  "react/jsx-no-bind": true,
  "react/jsx-no-comment-textnodes": true,
  "react/jsx-no-duplicate-props": true,
  "react/jsx-no-literals": true,
  "react/jsx-no-target-blank": true,
  "react/jsx-no-undef": true,
  "react/jsx-pascal-case": true,
  "react/jsx-sort-props": true,
  "react/jsx-space-before-closing": true,
  "react/jsx-tag-spacing": true,
  "react/jsx-uses-react": true,
  "react/jsx-uses-vars": true,
  "react/jsx-wrap-multilines": true,
  "react/no-array-index-key": true,
  "react/no-children-prop": true,
  "react/no-danger": true,
  "react/no-danger-with-children": true,
  "react/no-deprecated": true,
  "react/no-did-mount-set-state": true,
  "react/no-did-update-set-state": true,
  "react/no-direct-mutation-state": true,
  "react/no-find-dom-node": true,
  "react/no-is-mounted": true,
  "react/no-multi-comp": true,
  "react/no-redundant-should-component-update": true,
  "react/no-render-return-value": true,
  "react/no-set-state": true,
  "react/no-string-refs": true,
  "react/no-typos": true,
  "react/no-unescaped-entities": true,
  "react/no-unknown-property": true,
  "react/no-unused-prop-types": true,
  "react/no-unused-state": true,
  "react/no-will-update-set-state": true,
  "react/prefer-es6-class": true,
  "react/prefer-stateless-function": true,
  "react/prop-types": true,
  "react/react-in-jsx-scope": true,
  "react/require-default-props": true,
  "react/require-optimization": true,
  "react/require-render-return": true,
  "react/self-closing-comp": true,
  "react/sort-comp": true,
  "react/sort-prop-types": true,
  "react/style-prop-object": true,
  "react/void-dom-elements-no-children": true,
}
`;
